x/blob/types/payforblob.go

```diff
@@ -156,6 +179,41 @@ func CreateMultiShareCommitment(blobs ...*Blob) ([]byte, error) {
 	return merkle.HashFromByteSlices(commitments), nil
 }
 
+// ValidatePFBComponents performs basic checks over the components of one or more PFBs.
+func ValidatePFBComponents(nsIDs [][]byte, blobs [][]byte, shareVersions []uint32) error {
+	if len(nsIDs) != len(blobs) || len(nsIDs) != len(shareVersions) {
+		return ErrMismatchedNumberOfPFBComponent.Wrapf(
+			"namespaces %d blobs %d versions %d",
+			len(nsIDs), len(blobs), len(shareVersions),
+		)
+	}
+
+	if len(blobs) == 0 {
+		return ErrNoBlobs
+	}
+
+	for _, ns := range nsIDs {
+		err := ValidateBlobNamespaceID(ns)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, blob := range blobs {
+		if len(blob) == 0 {
+			return ErrZeroBlobSize
+		}
+	}
+
+	for _, v := range shareVersions {
+		if v != uint32(appconsts.ShareVersionZero) {
+			return ErrUnsupportedShareVersion
+		}
+	}
```

Comment:
[super optional] future proofing for when we add a new share version

```suggestion
	for _, shareVersion := range shareVersions {
		if !slices.Contains(appconsts.SupportedShareVersions, uint8(shareVersion)) {
			return ErrUnsupportedShareVersion
		}
	}```

Reply:
noice